src/main/java/httpRequest/HttpRequest.java

```diff
@@ -4,11 +4,25 @@
 public class HttpRequest implements IHttpRequest {
 
   String requestLine;
+  String method;
+  String path;
+  String version;
+  String parameters;
   HashMap<String, String> headers;
   byte[] body;
 
   public HttpRequest(String requestLine, HashMap<String, String> headers) {
     this.requestLine = requestLine;
+    this.method = requestLine.split(" ")[0];
```

Comment:
Is it okay to have your constructor call helper methods?? Like, I know you can do it. I just remember having teachers in school that would frown upon having your constructor call other methods, but I can't remember their reasoning anymore, so it might just be they were old and crazy!